validphys2/src/validphys/theorycovariance/construction.py

```diff
@@ -171,6 +171,37 @@ def combine_by_type(each_dataset_results_central_bytheory):
     )
     return process_info
 
+def process_starting_points(combine_by_type):
+    """Returns a dictionary of indices in the covariance matrix corresponding
+    to the starting point of each process."""
+    process_info = combine_by_type
+    running_index = 0
+    start_proc = defaultdict(list)
+    for name in process_info.theory:
+        size = len(process_info.theory[name][0])
+        start_proc[name] = running_index
+        running_index += size
+    return start_proc
+
+
+def covmap(combine_by_type, dataset_names):
```

Comment:
Instead of covmat please create a function `dataset_startingpoints` like the one for process.